### DISCUSSION TITLE
`NeighborSampler` dependencies

### DISCUSSION THREAD

**Discussion by john-como** [pyg-team/pytorch_geometric#7719] [Q&A] [answered]
Hello all. I am working to try and use mini-batching for a link level prediction task. I currently am using the class `LinkNeighborLoader` but am getting the following error : `'NeighborSampler' requires either 'pyg-lib' or 'torch-sparse'`. I have installed pyg-lib in the environment and reran all code, yet am still getting this issue.

Unsure why this is happening or how to resolve. Any pointers would be great! Thanks.

**Comment by wsad1** [comment, marked as answer]
Could you please share the following
1. Version of pyg and pyg-lib
2. The installation steps you followed

**Reply by john-como** [reply]
I have resolved, thank you @wsad1

**Comment by Romain-Nicolle** [comment]
Hi @john-como , could you please provide details on how you solved it ? I have a similar problem with the ```NeighborLoader```. 
I tried installing, uninstalling and installing again, with both pip and conda, with different versions etc. 
The installation appears to be OK, at least I do not have any error log.
But I also have the message ```'NeighborLoader' requires either 'pyg-lib' or 'torch-sparse'``` when I try to run a script.

Any pointers on how you managed to solve that would be a great help, thanks.

**Reply by LukeLIN-web** [reply]
I could provide a docker images. https://hub.docker.com/repository/docker/1263810658/pytorch-geometric/general

**Reply by Romain-Nicolle** [reply]
Thanks for your help. I will use that in the meantime, but I still will try to fix the installation on my end. With my specific versions of CUDA. But thanks for your assistance, greatly appreciated :-)

**Comment by WHU-LX** [comment]
Hi, I also had this problem but today I have solved it! Hope it can help u well!

1- Conda create a new environment, I create 'PyG' environment with Python 3.9.
2- conda install numy, conda install pandas(numpy is required before installing pytorch)
3- You should install PYTORCH first, as reconmmended in page 'https://pytorch.org/get-started/previous-versions/', conda install pytorch==2.1.0 torchvision==0.16.0 torchaudio==2.1.0 cpuonly -c pytorch. 
4- Then, install those packages IN ORDER: ①torch_scatter②torch_sparse③torch_cluster④torch_spline_conv(Step4)⑤torch-geometric(Step5)，You can download the wheel from website 'https://[data.pyg.org/whl/](https://data.pyg.org/whl/)', and the version should match your pytorch version, for me, it's pytorch==2.1.0.
5- Finally, pip install torch_geometric, as recommended in 'https://pytorch-geometric.readthedocs.io/en/latest/install/installation.html', because I use CPU, it will be different if u use GPU, CUDA, maybe, so, double check it.
6- If u use jupyter lab, remenber pip install ipykernel to change your environment.

Finally, it works. Wish it will help you too! Good luck!

Wuhan University, Gou Zhonghua's Lab.

**Reply by Romain-Nicolle** [reply]
Thanks for your reply @WHU-LX , greatly appreciated.

I have CUDA=11.6 so the latest pytorch version that I can use is the 1.13.1. But I will try that.
Up until now I had only tried using the GPU. So using the CPU only might do the trick, htanks for the suggestion.

Additionally I am using a Heterogeneous graph so I require the pyg-lib dependency. 
When installing all the dependencies, using the GPU, I still get the error message: "ImportError: 'NeighborSampler' requires either 'pyg-lib' or 'torch-sparse'"
But, I gav a look to the  source code of the package and I fear than the specific part that I need, NeighborLoader(..., disjoint = True, subgraph_type = 'induced', ...)  is not implemented yet for Heterogeneous Graph.

Either way, thank you for your assistance. And I will try the CPU only.

Best regards :-)
